ffi/SDL2_0.lua

```diff
@@ -377,11 +380,11 @@ function S.waitForEvent(sec, usec)
         else
             if pointers[0] and pointers[0].down then
                 slot = 0 -- left mouse button down
-                genTouchMoveEvent(event, slot, event.motion.x, event.motion.y)
+                genTouchMoveEvent(event, slot, event.motion.x * S.w / S.win_w, event.motion.y * S.h / S.win_h)
```

Comment:
That should be written like this. See [here](https://luapower.com/luajit-notes#luajit-assumptions).
```suggestion
                genTouchMoveEvent(event, slot, event.motion.x * S.w * (1 / S.win_w), event.motion.y * S.h * (1 / S.win_h))
```

Or possibly even `S.w * (1 / S.win_w)` as a local a bit higher. although for relatively simple math that might have a counterintuitive effect.

Reply:
Regardless of the explanation (as without measuring you don't really know anyway) I know I've seen performance significant performance differences in the past in basically this precise scenario using this precise method. But it would seem it's all the same here, meaning either something's subtly different or LuaJIT itself managed to improve on it since a few years ago. So just go with whatever you think is the most legible.

```
$ time ./luajit za.lua; time ./luajit zb.lua; time ./luajit zc.lua; time ./luajit zd.lua
50	50

real	0m10.635s
user	0m10.605s
sys	0m0.000s
50	50

real	0m10.635s
user	0m10.613s
sys	0m0.004s
50	50

real	0m10.626s
user	0m10.617s
sys	0m0.000s
50	50

real	0m10.628s
user	0m10.613s
sys	0m0.004s
```

```lua
-- za.lua
local w = 1404
local win_w = 2808
local h = 1872
local win_h = 3744

local res_x = 0
local res_y = 0

for i = 1,10000000000 do
    res_x = 100 * w * (1 / win_w)
    res_y = 100 * h * (1 / win_h)
end

print(res_x, res_y)
```

```lua
-- zb.lua
local w = 1404
local win_w = 2808
local h = 1872
local win_h = 3744

local res_x = 0
local res_y = 0

for i = 1,10000000000 do
    res_x = 100 * w / win_w
    res_y = 100 * h / win_h
end

print(res_x, res_y)
```

```lua
-- zc.lua
local w = 1404
local win_w = 2808
local h = 1872
local win_h = 3744

local res_x = 0
local res_y = 0

local scale_x = w / win_w
local scale_y = h / win_h

for i = 1,10000000000 do
    res_x = 100 * scale_x
    res_y = 100 * scale_y
end

print(res_x, res_y)
```

```lua
-- zd.lua
local w = 1404
local win_w = 2808
local h = 1872
local win_h = 3744

local res_x = 0
local res_y = 0

local S = {
    scale_x = w / win_w,
    scale_y = h / win_h,
}

for i = 1,10000000000 do
    res_x = 100 * S.scale_x
    res_y = 100 * S.scale_y
end

print(res_x, res_y)
```